.github/workflows/ci.yml

```
@@ -57,65 +65,70 @@ jobs:
+        run: dotnet build Consul.Test --configuration=Release -p:TreatWarningsAsErrors=true --framework=${{ matrix.framework }}
       - name: Run tests
         shell: bash
         run: |
-          cd Consul.Test
-          ./consul agent -dev -config-file test_config.json -log-file consul.log >consul-stdout.log 2>consul-stderr.log &
-          dotnet test --configuration=Release --logger "GitHubActions;report-warnings=false" --no-build -v=Normal
+          ./Consul.Test/consul agent -dev -config-file Consul.Test/test_config.json -log-file consul.log >consul-stdout.log 2>consul-stderr.log &
+          dotnet test Consul.Test --configuration=Release --logger "GitHubActions;report-warnings=false" --no-build -v=Normal --framework=${{ matrix.framework }}
       - name: Upload Consul logs
         if: failure()
         uses: actions/upload-artifact@v2
         with:
-          name: consul-logs-${{ matrix.os }}
-          path: Consul.Test/consul*.log
-      - name: Create NuGet package
-        if: runner.os == 'Windows'
-        run: |
-          if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
-            # WORKAROUND: Add letter prefix to ensure that MSBuild treats
-            # the suffix as a string. e.g. '0313071' will fail as an invalid
-            # version string, but 'G0313071' will succeeded. It looks like
-            # the parser does not like numbers with a leading zero.
-            $suffix = 'g' + $(git rev-parse --short HEAD)
-            $params = "--version-suffix", $suffix
-          }
+          name: consul-logs-${{ matrix.consul }}-${{ matrix.framework }}-${{ matrix.os }}
+          path: consul*.log
 
-          dotnet pack --configuration=Release --no-build @params Consul/Consul.csproj
+  package:
+    runs-on: windows-2019
+    needs: [Consul_AspNetCore, Consul]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Create NuGet package
+      run: |
+        if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
+          # WORKAROUND: Add letter prefix to ensure that MSBuild treats
+          # the suffix as a string. e.g. '0313071' will fail as an invalid
+          # version string, but 'G0313071' will succeeded. It looks like
+          # the parser does not like numbers with a leading zero.
+          $suffix = 'g' + $(git rev-parse --short HEAD)
+          $params = "--version-suffix", $suffix
+        }
+
+        dotnet pack --configuration=Release --output dist @params
 
-          if ("${{ github.ref }}" -like "refs/tags/v*") {
+        if ("${{ github.ref }}" -like "refs/tags/v*") {
             $tag = "${{ github.ref }}".SubString(11)
-            $expectedFile = "dist/consul/Consul.$tag.nupkg"
+            $expectedFile = "dist/Consul.$tag.nupkg"
             # Check whether the tag and the package version match together
             if (-not (Test-Path -Path $expectedFile)) {
                 echo "Expected file $expectedFile doesn't exist"
```

We should output a proper workflow error here (see the [docs](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message))

```suggestion
                echo "Expected file $expectedFile doesn't exist"
                echo "::error ::Expected file $expectedFile doesn't exist"
```

---

Done